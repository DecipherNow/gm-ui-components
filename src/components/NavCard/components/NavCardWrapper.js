import styled from "styled-components";
import { mix, darken } from "polished";

import { spacingScale } from "style/styleFunctions";
import {
  OPACITY_15,
  COLOR_BACKGROUND_B,
  COLOR_BACKGROUND_C,
  COLOR_INTENT_HIGHLIGHT,
  FONT_WEIGHT_REGULAR,
  BORDER_RADIUS_BASE,
  FONT_STACK_BASE
} from "style/styleVariables";

const COLOR_TAB_BACKGROUND_BASE = mix(
  OPACITY_15,
  COLOR_BACKGROUND_B,
  COLOR_BACKGROUND_C
);

const TAB_WIDTH_BASE = "1%";

// Note: Edge requires the overflow: hidden property to maintian
// equal sized cards. flex-basis is not sufficient!
const NavCardWrapper = styled.div`
  font-weight: ${FONT_WEIGHT_REGULAR};
  font-family: ${FONT_STACK_BASE};
  flex: 1 1 ${TAB_WIDTH_BASE};
  background-color: ${COLOR_TAB_BACKGROUND_BASE};
  border-radius: ${BORDER_RADIUS_BASE};
  min-height: ${spacingScale(8)};
  position: relative;
  transition: all 0.15s ease;
  display: flex;
  flex-direction: column;
  align-items: stretch;
  min-height: ${spacingScale(10)};
  margin: ${spacingScale(0.25)};

  &:after {
    content: "";
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    border-radius: 4px;
    border: 2px solid ${COLOR_INTENT_HIGHLIGHT};
    opacity: 0;
  }
  &:focus,
  &:hover {
    background-color: ${darken(0.06, COLOR_TAB_BACKGROUND_BASE)};
  }
`;

export default NavCardWrapper;
