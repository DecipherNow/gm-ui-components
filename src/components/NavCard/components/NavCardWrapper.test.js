import React from "react";
import { shallow } from "enzyme";
import NavCardWrapper from "./NavCardWrapper";

describe("NavCardWrapper", () => {
  it("matches snapshot", () => {
    const aNavCardWrapper = shallow(<NavCardWrapper />);
    expect(aNavCardWrapper).toMatchSnapshot();
  });
});
