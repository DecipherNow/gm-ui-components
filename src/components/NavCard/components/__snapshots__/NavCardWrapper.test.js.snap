// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavCardWrapper matches snapshot 1`] = `
.c0 {
  background-color: #252525;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen","Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue",sans-serif;
  -webkit-flex: 1 1 1%;
  -ms-flex: 1 1 1%;
  flex: 1 1 1%;
  border-radius: 4px;
  min-height: 64px;
  position: relative;
  -webkit-transition: all 0.15s ease;
  transition: all 0.15s ease;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  min-height: 80px;
  margin: 2px;
}

.c0:after {
  content: "";
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: 4px;
  border: 2px solid #0aab2a;
  opacity: 0;
}

.c0:focus,
.c0:hover {
  background-color: #161616;
}

<div
  className="c0"
/>
`;
